server/mqtt.go

```diff
@@ -85,6 +85,7 @@ const (
 	mqttConnAckRCServerUnavailable           = byte(0x3)
 	mqttConnAckRCBadUserOrPassword           = byte(0x4)
 	mqttConnAckRCNotAuthorized               = byte(0x5)
+	mqttConnAckRCQoS2WillRejected            = byte(0x10)
```

Comment:
@kozlovic I added this RC, non-continuos with the rest of them because again, hopefully it's temporary; otherwise the CONNECT was dropped and EOF returned instead of a CONNACK with RC. The new test revealed it.